client/src/Form.js

```diff
@@ -0,0 +1,70 @@
+import React, {useState} from "react";
+import "./App.css";
+
+const Form = ({setVideos}) =>{
+    const [title, setTitle] = useState("");
+    const [url, setUrl] = useState("");
+    const [rating, setRating] = useState(0);
+
+    const submitForm = (e) => {
+      e.preventDefault();
+      let newVideo = {
+        id: Date.now(),
+        title: title,
+        url: url,
+        rating: rating
+      }
+      function matchYoutubeUrl(url) {
+        var p =
+          /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
```

Comment:
Always use `let` or `const` instead of `var`.

Also, I can see that you have used a regular expression in this particular code. Can you please explain why have you used this regular expression? What is it doing and what are the benefits and disbenefits of using a regex here?

Reply:
In this case I have used the regex to help validate the URL that is submitted with the form as a valid YouTube link.